From: SeongJae Park <sj@kernel.org>
Date: Wed, 7 Jan 2026 17:45:17 -0800
Subject: [PATCH] ==== config_damon_debug_sanity ====

mm/damon: add optional debugging-purpose sanity checks

DAMON code has a few assumptions that can be critical if violated.
Validating the assumptions in code can be useful at finding such
critical bugs.  I was actually adding some such additional sanity checks
in my personal tree, and those were useful at finding bugs that I made
during the development of new patches.  We also found [1] sometimes the
assumptions are misunderstood.  The validation can work as a good
documentation for such cases.

Add some of such debugging purpose sanity checks.  Because those
additional checks can impose more overhead, make those only optional via
new config, CONFIG_DAMON_DEBUG_SANITY, that recommended for only
development and test setups.  And as recommended, enable it for DAMON
kunit tests and selftests.

[1] https://lore.kernel.org/20251231070029.79682-1-sj@kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_4mbsgzz3 | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_4mbsgzz3

diff --git a/hkml_cv_bogus/hkml_cv_bogus_4mbsgzz3 b/hkml_cv_bogus/hkml_cv_bogus_4mbsgzz3
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

